modules/standard/FFTW.chpl

```diff
@@ -20,40 +20,71 @@
+  :mod:`FFTW_MT` module which provides support for calls to the
+  multi-threaded FFTW implementation.
+
+
+  To use this module:
+
+  1. Ensure that FFTW (version 3) is installed on your system and that
+     the header and library files (e.g., fftw3.h, libfftw3.*) are
+     either installed in a standard system location or that your C
+     compiler's environment variables are set up to find them.
+
+  2. Add ``use FFTW;`` to your Chapel code.
+
+  3. Compile and run your Chapel program as usual.
 
-  This module defines a Chapel wrapper around key routines from FFTW
-  version 3.  Over time, the intention is to expand this support to
-  additional routines, prioritizing based on user feedback.
 
   As in standard FFTW usage, the flow is to:
 
-  1) Create plan(s) using the plan_dft*() routines.
+  1. Create plan(s) using the :proc:`plan_dft* <plan_dft>` routines.
+
+  2. Execute the plan(s) one or more times using :proc:`execute`.
 
-  2) Execute the plan(s) using the execute() routine.
+  3. Destroy the plan(s) using :proc:`destroy_plan`.
 
-  3) Destroy the plan(s) using the destroy_plan() routine.
+  4. Call :proc:`cleanup`.
 
-  4) Call cleanup().
 
-  At present, clients of the FFTW module must specify fftw3.h and -I,
-  -L, -l flags sufficient to specify the required libraries and find
-  the files on the 'chpl' command line.  We are currently
-  investigating techniques to streamline this process.
+  Note that each of the Chapel :proc:`plan_dft* <plan_dft>` routines
+  support both `in-place` and `out-of-place` versions of the
+  transforms, where the former versions use a single array for both
+  input and output, and the latter use two distinct arrays.
+  
+  In future versions of this module, we anticipate improving the
+  plan_dft*() interfaces to make better use of Chapel features and
+  move further away from C-isms (like the overloaded role of `flags`
+  and the use of C-based types).  Such features are expected to take
+  advantage of Chapel's support for default argument values and
+  keyword-based argument passing.  We are also thinking about changing
+  the interface for the in-place routines to use array slicing rather
+  than separate arguments for the array and domain.
 */
 
-// TODOs for Brad (in priority order):
-//
-// - Can we get a form of use/requires in that will simplify the
-//   command-line?
+// Possible TODOs:
 //
 // - How do we feel about FFTW_ALLCAPS names given that the routines
 //   themselves don't have fftw_ prefixes, that they're defined as
-//   c_int's, and that they're all-caps?
+//   c_int's, and that they're all-caps?  What if we were to replace
+//   these with bool/enum arguments representing the separate planning
+//   aspects that are overloaded into 'flags'?  This would also allow
+//   us to move away from c_ints (which could be done in other ways
+//   as well.
+//
+// - It seems that rather than passing in a domain and an array for
```

Comment:
Brad: It seems that the comment that starts "It seems that rather than ..." was truncated.